src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonerService.kt

```diff
@@ -53,10 +53,15 @@ class PrisonerService(
   }
 
   private fun isPrisonerInTemporaryLocation(levels: List<PrisonerHousingLevelDto>): Boolean {
-    if (levels.isNotEmpty()) {
-      return TransitionalLocationTypes.contains(levels[0].code)
-    }
-    return false
+    return (levels.isNotEmpty() && TransitionalLocationTypes.contains(levels[0].code))
+  }
+
+  fun isPrisonerInTAPTemporaryLocation(prisonerHousingLocation: PrisonerHousingLocationsDto?): Boolean {
```

Comment:
is this the correct naming?

Reply:
sessionTemplates = if (sessionTemplates.any { it.isTapSession } && prisonerService.isPrisonerInTAPTemporaryLocation(prisonerHousingLocation)) {
      sessionTemplates.filter { it.isTapSession }
    } else {
      sessionTemplates.filter {
        isAvailableToLocation(it, sessionTemplates, prisonerId, prisonCode, prisonerHousingLocation)
          .and(isAvailableToCategory(it, sessionTemplates, prisoner.category))
          .and(isAvailableToIncentiveLevel(it, sessionTemplates, prisoner.incentiveLevel))
      }
    }

I think all the logic should be data driven not a hard coded connection, so the location group locations should have TAP in it, set in admin

Reply:
Updated to be data driven as suggested with the includeLocationGroups flag.